packages/platform-sdk-profiles/src/environment/services/exchange-rate-service.ts

```diff
@@ -36,17 +46,43 @@ export class ExchangeRateService {
 			return;
 		}
 
+		const exchangeCurrency: string = profile.settings().get(ProfileSetting.ExchangeCurrency) || "BTC";
+		if (!this.shouldSync(exchangeCurrency)) return;
+
 		const marketService = MarketService.make(
 			profile.settings().get(ProfileSetting.MarketProvider) || "coingecko",
 			container.get(Identifiers.HttpClient),
 		);
-
-		const exchangeCurrency: string = profile.settings().get(ProfileSetting.ExchangeCurrency) || "BTC";
 		const exchangeRate = await marketService.dailyAverage(currency, exchangeCurrency, +Date.now());
 
+		this.#lastSynced[exchangeCurrency] = DateTime.make();
+		const date = DateTime.make().format("YYYY-MM-DD");
+
 		for (const wallet of wallets) {
-			wallet.data().set(WalletData.ExchangeCurrency, exchangeCurrency);
-			wallet.data().set(WalletData.ExchangeRate, exchangeRate);
+			this.updateWalletExchangeData(wallet, exchangeCurrency, exchangeRate, date);
```

Comment:
This is a promise but not awaited. You should `Promise.all` or `Promise.allSettled` here to get this done fast without having unresolved promises.

Reply:
@faustbrian  was incorrectly set as promise. `data().get` and `data.set` are sync operations. Fixed it